comp/core/profiler/impl/profiler.go

```diff
@@ -0,0 +1,299 @@
+				},
+				{
+					// CPU profile
+					name: service + "-cpu.pprof",
+					path: fmt.Sprintf("/profile?seconds=%d", seconds),
+				},
+				{
+					// 2nd heap profile
+					name: service + "-2nd-heap.pprof",
+					path: "/heap",
+				},
+				{
+					// mutex profile
+					name: service + "-mutex.pprof",
+					path: "/mutex",
+				},
+				{
+					// goroutine blocking profile
+					name: service + "-block.pprof",
+					path: "/block",
+				},
+				{
+					// Trace
+					name: service + ".trace",
+					path: fmt.Sprintf("/trace?seconds=%d", seconds),
+				},
+			} {
+				b, err := get(prof.path)
+				if err != nil {
+					return err
+				}
+				pdata[prof.name] = b
+			}
+			return nil
+		}
+	}
+
+	agentCollectors := map[string]agentProfileCollector{}
+	if p.coreAgentEnabled() {
+		agentCollectors["core"] = serviceProfileCollector(tcpGet("expvar_port"), seconds)
+	}
+
+	if p.securityAgentEnabled() {
+		agentCollectors["security-agent"] = serviceProfileCollector(tcpGet("security_agent.expvar_port"), seconds)
+	}
+
+	if p.processAgentEnabled() {
+		agentCollectors["process"] = serviceProfileCollector(tcpGet("process_config.expvar_port"), seconds)
+	}
+
+	if p.apmEnabled() {
+		traceCpusec := p.apmTraceSeconds(seconds)
+		agentCollectors["trace"] = serviceProfileCollector(tcpGet("apm_config.debug.port"), traceCpusec)
+	}
+
+	if p.sysProbeEnabled() {
+		probeUtil, probeUtilErr := procnet.GetRemoteSystemProbeUtil(p.sysProbeCfg.GetString("system_probe_config.sysprobe_socket"))
+
+		if !errors.Is(probeUtilErr, procnet.ErrNotImplemented) {
+			sysProbeGet := func() pprofGetter {
+				return func(path string) ([]byte, error) {
+					if probeUtilErr != nil {
+						return nil, probeUtilErr
+					}
+
+					return probeUtil.GetPprof(path)
+				}
+			}
+
+			agentCollectors["system-probe"] = serviceProfileCollector(sysProbeGet(), seconds)
+		}
+	}
+
+	var errs error
+	for name, callback := range agentCollectors {
+		if err := callback(name); err != nil {
+			errs = multierror.Append(errs, fmt.Errorf("error collecting %s agent profile: %v", name, err))
+		}
+	}
+
+	return pdata, errs
+}
+
+func (p profiler) setProfilerSetting(settingName string, newValue int, fb flaretypes.FlareBuilder) func() {
+	oldValue, err := p.settingsComponent.GetRuntimeSetting(settingName)
+	if err != nil {
+		_ = fb.Logf("Unable to access %s setting, ignoring value provided by flare args: %v", settingName, err)
+
+		return nil
+	} else if newValue <= 0 || newValue == oldValue {
+		return nil
+	}
+
+	err = p.settingsComponent.SetRuntimeSetting(settingName, newValue, model.SourceAgentRuntime)
+	if err != nil {
+		_ = fb.Logf("Unable to set the %s: %v", settingName, err)
+	}
+
+	return func() {
+		nErr := p.settingsComponent.SetRuntimeSetting(settingName, oldValue, model.SourceAgentRuntime)
+		if nErr != nil {
+			_ = fb.Logf("Unable to reset the %s back to its original value: %v", settingName, nErr)
+		}
+	}
+}
+
+func (p profiler) fillFlare(fb flaretypes.FlareBuilder) error {
```

Comment:
Would it make sense to add a comment that this is a special case and is only used when the flare is collected via RC, rather than via the `flare` sub-command? Otherwise it doesn't make sense to change runtime settings directly here and took me a while to figure out that this code path is not used by the subcommand.

Reply:
Absolutely 👍 👍 
Will also add documentation to ReadProfileData along similar lines